compare50/__main__.py

```diff
@@ -61,20 +62,41 @@ def exclude(self, pattern):
     def _get(self, path, preprocessor, is_archive=False):
         path = pathlib.Path(path)
 
+        # Ask lib50 which file(s) in path should be included
         if path.is_file():
+            # lib50.files operates on a directory
+            # So create a tempdir if the path is just a file
             with tempfile.TemporaryDirectory() as dir:
                 (pathlib.Path(dir) / path.name).touch()
                 included, excluded = lib50.files(self.patterns, root=dir)
             path = path.parent
         else:
             included, excluded = lib50.files(self.patterns, require_tags=[], root=path)
 
-        decodable_files = sorted(file_path for file_path in included if self._is_valid_utf8(path / file_path))
-
-        if not decodable_files:
-            raise _api.Error(f"Empty submission: {path}")
+        # Filter out any non utf8 files
+        undecodable_files = []
+        decodable_files = []
+        for file_path in included:
```

Comment:
I feel like we should factor this out.

```
def partition(vals, pred):
    true = set()
    false = set()
    for val in  vals:
        (true if pred(val) else false).add(val)
    return true, false

# ...

decodable, undecodable = partition(included, lambda fp: self._is_valid_utf8(path / fp))
```

and then ditto for large/small later.